Quadriga/src/main/java/edu/asu/spring/quadriga/service/passthroughproject/impl/XMLReader.java

```diff
@@ -68,14 +68,17 @@ public String getNetwork(String xml) {
      * @param tagName
      * @return The id for the given tag.
      */
-    private String getTagId(Document document, String tagName) {
+    private String getTagId(Document document, String tagName) throws DocumentParserException {
 
         Node node = document.getElementsByTagName(tagName).item(0);
         if (node == null) {
             return null;
         }
         NamedNodeMap nodeAttributeMap = node.getAttributes();
         Node idNode = nodeAttributeMap.getNamedItem("id");
+        if(idNode == null) {
+            throw new DocumentParserException("Please provide a " + tagName + " id as a part of the XML");
```

Comment:
And actually, I don't think this is correct. Please check the logic, but I think a user can create a project or workspace by submitting no ids.